catkit2/services/oceanoptics_spectrometer/oceanoptics_spectrometer.py

```diff
@@ -0,0 +1,187 @@
+'''
+This module contains a service for Ocean Optics Spectrometers.
+'''
+
+import threading
+
+from seabreeze.spectrometers import Spectrometer
+from seabreeze.spectrometers import SeaBreezeError
+
+from catkit2.testbed.service import Service
+
+
+class OceanOpticsSpectrometer(Service):
+    '''
+    Service for Ocean Optics Spectrometer.
+
+    This service is a wrapper around the python seabreeze package.
+    It provides a simple interface to control the spectrometer and acquire spectra.
+
+    Attributes
+    ----------
+    spectrometer : Spectrometer
+        The spectrometer to control.
+    serial_number : str
+        The serial number of the spectrometer.
+    model : str
+        The model of the spectrometer.
+    pixels_number : int
+        number of pix in the spectrum.
+    wavelengths : NDArray[numpy.float_]
+        A NDArray with the wavelengths of the spectrometer.
+    spectra : DataStream
+        A data stream to submit spectra from the spectrometer.
+    should_be_acquiring : threading.Event
+        An event to signal whether the spectrometer should acquire a spectrum.
+    NUM_FRAMES : int
+        The number of frames to allocate for the data streams.
+
+    Methods
+    -------
+    open()
+        Open the service.
+    main()
+        The main function of the service.
+    close()
+        Close the service.
+    start_acquisition()
+        Start the acquisition.
+    take_one_spectrum()
+        Get a spectrum from the spectrometer.
+    exposure_time()
+        Set the exposure time of the spectrometer.
+    '''
+    NUM_FRAMES = 20
+
+    def __init__(self):
+        '''
+        Create a new OceanOpticsSpectrometer service.
+        '''
+        super().__init__('oceanoptics_spectrometer')
+
+        self.serial_number = None
+        self.spectrometer = None
+        self.model = None
+        self.pixels_number = None
+
+        self.wavelengths = None
+        self.spectra = None
```

Comment:
We never initialize attributes that hold data streams (maybe with a few exceptions). I have no clue if that's the right way to go, but at this point of time in the project it's probably good to keep things consistent as much as possible.

Reply:
ok